src/components/networkSwitcher.vue

```diff
@@ -0,0 +1,255 @@
+                {{network.url}}
+              </div>
+            </div><ae-check class="network-switcher__action"
+                            v-model="networkUrl"
+                            :value="network.url"
+                            type="radio"
+                            @change="catchNetworkName(network.name, network.url)"/>
+          </ae-list-item>
+        </div>
+      </ae-list>
+      <div class="hendler-wrapper">
+        <error class="error" :name="networkName"  @back="closeError" @closeNetwork="showNetworkList"  v-if="isError"></error>
+        <loader-item :name="networkName"  v-if="isNetworkChanging" ></loader-item>
+      </div>
+      <ae-toolbar slot="footer"  v-if="isDisplaying ">
+        <ae-button class="network-info__connect" face="flat" @click="showForm()">Connect to other network</ae-button>
+      </ae-toolbar>
+    </ae-card>
+  </div>
+</template>
+
+<script>
+
+import { mapState } from 'vuex'
+import {
+  AeModal
+} from '@aeternity/aepp-components'
+import LoaderItem from './loaderItem'
+import Error from './error'
+
+import {
+  AeCard,
+  AeList,
+  AeListItem,
+  AeCheck,
+  AeToolbar,
+  AeButton,
+  AeDivider,
+  AeDropdown,
+  AeIcon,
+  AeLoader,
+  AeFilterSeparator,
+  AePanel,
+  AeInput
+} from '@aeternity/aepp-components-3'
+
+export default {
+  name: 'network-switcher',
+  components: {
+    AeModal,
+    AeCard,
+    AeList,
+    AeListItem,
+    AeCheck,
+    AeToolbar,
+    AeButton,
+    AeDivider,
+    AeDropdown,
+    AeIcon,
+    AeLoader,
+    AeFilterSeparator,
+    AePanel,
+    AeInput,
+    LoaderItem,
+    Error
+  },
+  data () {
+    return {
+      view: null,
+      modalVisible: false,
+      networkUrl: this.$store.state.epochUrl,
+      networkName: undefined,
+      serverNetworks: [
+        {
+          name: 'Roma',
+          url: 'https://sdk-mainnet.aepps.com/'
+        },
+        {
+          name: 'sdk-edgenet.aepps',
+          url: 'https://sdk-edgenet.aepps.com/'
+        },
+        {
+          name: 'sdk-testnet.aepps',
+          url: 'https://sdk-testnet.aepps.com/'
+        }
+      ],
+      isNetworkChanging: false,
+      isError: false
+    }
+  },
+  methods: {
+    closeError (e) {
+      this.isError = e
+    },
+    showNetworkList (isVisible) {
+      this.$emit('showList', isVisible)
+    },
+    async changeNetwork () {
+      this.isNetworkChanging = true
+      this.$store.commit('changeNetworkUrl', this.networkUrl)
+      this.$store.commit('blocks/resetState')
+      this.$store.commit('accounts/resetState')
+      this.$store.commit('transactions/resetState')
+
+      await Promise.all([
+        this.$store.dispatch('blocks/height'),
+        this.$store.dispatch('getNodeStatus')
+      ]).then(success => {
+        this.$store.commit('clearError')
+        this.showNetworkList(false)
+      }).catch(err => {
+        this.isError = true
+        this.isNetworkChanging = false
+        this.$store.commit('catchError', err)
+      })
```

Comment:
The common code between [`connectionForm.vue`](https://github.com/aeternity/aepp-blockchain-explorer/pull/136/files#diff-e8e824a76dea39c9ec70abc78a5cfac4R70) and this component should be extracted to Vuex.